recipes/redis-plus-plus/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class RedisPlusPlusConan(ConanFile):
+    name = "redis-plus-plus"
+    homepage = "https://github.com/sewenew/redis-plus-plus"
+    description = "Redis client written in C++"
+    topics = ("conan", "database", "redis", "client", "tls")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "with_tls": [True, False], "with_cxx": [11, 17]}
+    default_options = {"shared": False, "fPIC": True, "with_tls": False, "with_cxx": 11}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+    def validate(self):
+        if self.settings.compiler == "Visual Studio" and self.optioins.shared:
```

Comment:
```suggestion
            del self.options.fPIC

    def validate(self):
        if self.settings.compiler == "Visual Studio" and self.options.shared:
```

Reply:
If a thread could be opened to discuss what should go to `validate()` and what should stay in `configure()`, it would be nice, because I don't think that moving such verifications in `validate()` is a good thing.

Reply:
https://github.com/conan-io/conan-center-index/issues/4261